### PR TITLE
Fix up GetProcAddr handling in object_tracker, threading and parameter_validation layers

### PR DESCRIPTION
**Pull request by mark-lunarg**
Same as the previous PR, prolly one more coming for the rest of the layers.

### CODE REVIEW

File: layers/threading.cpp [hunk 1]
```diff
@@ -241,62 +234,28 @@ VKAPI_ATTR VkResult VKAPI_CALL EnumerateDeviceExtensionProperties(VkPhysicalDevi
...
 
 VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL GetInstanceProcAddr(VkInstance instance, const char *funcName) {
-    PFN_vkVoidFunction addr;
-    layer_data *my_data;
-
-    addr = layer_intercept_instance_proc(funcName);
-    if (!addr) addr = layer_intercept_proc(funcName);
-    if (addr) {
-        return addr;
+    const auto &item = name_to_funcptr_map.find(funcName);
```
Comment (chrisforbes) [inline layers/threading.cpp]: & isnt necessary here, item is an iterator.

File: layers/parameter_validation.cpp [hunk 2]
```diff
@@ -5742,36 +5742,29 @@ VKAPI_ATTR void VKAPI_CALL SetHdrMetadataEXT(VkDevice device, uint32_t swapchain
...
+    const auto &item = name_to_funcptr_map.find(funcName);
+    if (item != name_to_funcptr_map.end()) {
+        return reinterpret_cast<PFN_vkVoidFunction>(item->second);
+    }
 
-    if (!dev_data->dispatch_table.GetDeviceProcAddr) return nullptr;
-    return dev_data->dispatch_table.GetDeviceProcAddr(device, funcName);
+    layer_data *device_data = GetLayerDataPtr(get_dispatch_key(device), layer_data_map);
+    // LUGMAL -- what does ref do here, anything?
+    auto &table = device_data->dispatch_table;
```
Comment (chrisforbes) [inline layers/parameter_validation.cpp]: Layer data holds the dispatch table by value -- and it's big, so you don't want to copy it. `const &` would be better.

File: layers/parameter_validation.cpp [hunk 3]
```diff
@@ -5742,36 +5742,29 @@ VKAPI_ATTR void VKAPI_CALL SetHdrMetadataEXT(VkDevice device, uint32_t swapchain
...
+    return table.GetDeviceProcAddr(device, funcName);
 }
 
 VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL GetInstanceProcAddr(VkInstance instance, const char *funcName) {
-
-    PFN_vkVoidFunction addr = layer_intercept_proc(funcName);
-    if (addr) return addr;
-
-    assert(instance);
+    const auto &item = name_to_funcptr_map.find(funcName);
```
Comment (chrisforbes) [inline layers/parameter_validation.cpp]: More ref-to-iterator... this is all over the place.

**Comment by chrisforbes** [comment]
:+1: @mark-lunarg, looks good; just minor nits above.

**Comment by mark-lunarg** [comment]
Made @chrisforbes suggested fixes (thanks!) and pushed to Master.

**Pull request merged**